inference/core/workflows/core_steps/models/foundation/paligemma.py

```diff
@@ -0,0 +1,109 @@
+from typing import Any, Dict, List, Literal, Optional, Tuple, Type, Union
+
+from pydantic import AliasChoices, ConfigDict, Field
+
+from inference.core.entities.requests.paligemma import PaliGemmaInferenceRequest
+from inference.core.managers.base import ModelManager
+from inference.core.workflows.core_steps.common.utils import (
+    attach_parent_info,
+    attach_prediction_type_info,
+    load_core_model,
+)
+from inference.core.workflows.entities.base import OutputDefinition
+from inference.core.workflows.entities.types import (
+    BATCH_OF_PARENT_ID_KIND,
+    BATCH_OF_STRING_KIND,
+    STRING_KIND,
+    FlowControl,
+    StepOutputImageSelector,
+    WorkflowImageSelector,
+    WorkflowParameterSelector,
+)
+from inference.core.workflows.prototypes.block import (
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+from inference.models.paligemma.paligemma import PaliGemma
+
+LONG_DESCRIPTION = """
+Paligemma block TODO
+"""
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "short_description": "Run Paligemma model.",
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "model",
+        }
+    )
+    type: Literal["Paligemma"]
```

Comment:
I think this should be `PaliGemmaModel` (all model blocks should end in `Model` IMO)